### ISSUE TITLE
Web -Connect Bank Account - Blank modal after KBA check

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#14477]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Prerequisites: Go to OldDot, log in with a new applause.expensifail account, verify you see a payment type for WALLET with a deposit status of "Silver". Do not add a personal bank account. If already added, delete it.

1. Navigate to NewDot, log in with the previously created account and open a 1:1 conversation with any user
2. Click/tap on the + icon and select the "Send Money" option
3. Enter any amount and continue
4. Click on the button "Pay with Expensify"
5. Click Bank account and add a personal bank account using Wells Fargo credentials (Repro with Fidelity)
6. Once the bank account has been added, attempt to send money again
7. Verify that you're taken directly to the KYC verification
8. In the additional details form, enter first name "Alberta", last name "Bobbeth" and SSN as "3333". Enter any data for the other fields.
9. After getting error regarding SSN, don't change the SSN data and click Save & continue
10. KBA flow is triggered (flow to verify the identity via questions)
11. Select the first answer for every question (important otherwise the KBA check will fail)

## Expected Result:
User should proceed to the Onfido verification flow

## Actual Result:
Blank modal is displayed.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [x] Web / Chrome

**Version Number:** 1.2.57.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/3087583

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/214115438-d55ae52e-f851-4c59-919c-69909587cdbc.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0159b9a6a0a768f3db</li>
        <li>Upwork Job ID: 1623588390235365376</li>
        <li>Last Price Increase: 2023-02-09</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
@johncschuster this issue was created 2 weeks ago. Are we close to a solution? Let's make sure we're treating this as a top priority. Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
@johncschuster this issue was created 2 weeks ago. Are we close to a solution? Let's make sure we're treating this as a top priority. Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
@johncschuster this issue was created 2 weeks ago. Are we close to a solution? Let's make sure we're treating this as a top priority. Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
Triggered auto assignment to @neil-marcellini (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by johncschuster** [comment]
My bad on missing the ping here. Sorry team.

**Comment by neil-marcellini** [comment]
I'm able to reproduce this using a new @expensifail.com account, which automatically comes with a silver wallet. After I add a bank account in step 6 it sends me back to Plaid which seems like another error. I manually exited that flow and moved to step 7.

I'm not sure if this can be worked on externally so I'll ask in Slack. I have a full load of daily issues right now so if it's not external I'm going to ask if someone else can work on it.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0159b9a6a0a768f3db

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @thesahindia (`Internal`)

**Comment by MariaHCD** [comment]
Making this internal and taking over for @neil-marcellini (Slack thread [here](https://expensify.slack.com/archives/C03TQ48KC/p1675889812222889))

**Comment by MariaHCD** [comment]
Looking into this, it looks like the KYC wallet flow is not being triggered at all (related issue: https://github.com/Expensify/App/issues/14945)


https://user-images.githubusercontent.com/12268372/217808362-ffc7a841-4326-4319-ad0e-55f1f88d8a5d.mov

Investigating further.

**Comment by MariaHCD** [comment]
Looks like we removed the Onyx key `WALLET_ADDITIONAL_DETAILS_DRAFT` [here](https://github.com/Expensify/App/pull/13170/files#diff-2ed984677d91a13142dfc8ea0debc7002f6f8c8821e7c460e7a2b85075535e64L268-L269) but it is still being used for the Onfido step:

https://github.com/Expensify/App/blob/d9a126dba03ea2956176e33ff1c6d869a2653d03/src/pages/EnablePayments/EnablePaymentsPage.js#L82-L83

https://github.com/Expensify/App/blob/d9a126dba03ea2956176e33ff1c6d869a2653d03/src/pages/EnablePayments/OnfidoPrivacy.js#L48-L54

I think we don't need to use the additionalDetailsDraft at the OnfidoStep, we can just update the API command `OpenOnfidoFlow` to fetch these on the backend instead of relying on the frontend to pass these two parameters.

**Comment by MariaHCD** [comment]
Web and App PRs are still WIP but it looks like removing the usage of the Onyx key walletAdditionalDetailsDraft from the OnfidoStep worked 🎊 

1. KYC is triggered when attempting to send money with a silver wallet

https://user-images.githubusercontent.com/12268372/218100476-1f198da8-690d-4cbd-9d00-a1e2b88db21a.mov

2. KYC is triggered when attempting to pay an IOU with a silver wallet

https://user-images.githubusercontent.com/12268372/218100834-a08b57c1-fff0-4549-9fe6-f2b533b5404b.mov

Next steps for me:
- Ensure the KBA flow works as expected and the user can proceed to activate their wallet
- Fix the `Continue` button on the OnfidoPrivacy page to be fixed at the bottom of the page instead of in the middle
- Fix issue where adding a PBA as part of sending money redirects to the payments page as well as re-triggers Plaid.

**Comment by MariaHCD** [comment]
Resuming work on [the](https://github.com/Expensify/App/pull/14988) [PRs](https://github.com/Expensify/Web-Expensify/pull/36435) this week

**Comment by MelvinBot** [comment]
@johncschuster, @MariaHCD, @thesahindia Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MariaHCD** [comment]
Both [Web](https://github.com/Expensify/Web-Expensify/pull/36435) and [App](https://github.com/Expensify/App/pull/14988) PRs were merged and deployed to production. Created a separate issue to look into why Plaid is being re-triggered after adding a PBA: https://github.com/Expensify/App/issues/15202

(Just realized I didn't tag @thesahindia for review on the App PR, sorry!)

I think we should be all good here.
